packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -193,7 +193,7 @@ export default function jsx({ settings, logging }: AstroPluginJSXOptions): Plugi
 
 			const { mode } = viteConfig;
 			// Shortcut: only use Astro renderer for MD and MDX files
-			if (id.includes('.mdx') || isMarkdownFile(id, { criteria: 'includes' })) {
+			if (id.endsWith('.mdx') || isMarkdownFile(id, { criteria: 'endsWith' })) {
```

Comment:
Do we need the `isMarkdownFile` call here? Looks like the `!JSX_EXTENSIONS.has(path.extname(id))` guard above would always make this false.

Reply:
This is correct. I will remove it.